src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatSelection.cs

```diff
@@ -56,9 +56,9 @@ private bool TryExecuteCommand(FormatSelectionCommandArgs args, CommandExecution
 
                 var formattingSpan = selection[0].Span.ToTextSpan();
 
-                Format(args.TextView, document, formattingSpan, context.WaitContext.UserCancellationToken);
+                Format(args.TextView, document, formattingSpan, context.WaitContext.UserCancellationToken).Wait();
```

Comment:
this seems bad.  You are blocking . It's not cancellable.  How is progress shown to the user?

Reply:
reverted